staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go

```diff
@@ -35,12 +35,17 @@ import (
 
 var (
 	explainLong = templates.LongDesc(i18n.T(`
-		Describe fields and structure of various resources.
+		Describe (sub-)fields and structure of various resources.
 
-		This command describes the fields associated with each supported API resource.
-		Fields are identified via a simple JSONPath identifier:
+		This command describes the (sub-)fields associated with each supported API resource.
+		Fields are identified via a simple JSONPath accessor:
```

Comment:
This should probably be selector or query instead of either identifier or accessor, because this is the language that the RFC uses.

Reply:
The issue is that it’s not treated as a JSONPath query. It resembles JSONPath in that the field names are separated by dots. Otherwise the syntax isn’t JSONPath.

Describing it as a JSONPath selector or query would be a clear improvement if this were how the value were treated. But since it isn’t, it’s not much of a help in terms of aligning this part of the help text with the actual behavior.

As an example, JSONPath would allow “ explain storage.spec..proxy”. But it’s rejected here (error: field "" does not exist).

In my view, references to JSONPath suggest that the value will be treated as a JSONPath expression. It may be clearer to remove references to JSONPath, or to allow the use of JSONPath queries.

I can retest with the latest kubectl and post specific examples of rejected JSONPath expressions if that would be helpful.

Reply:
Ah, interesting, ok let me take a look at the backend code for this. This is more likely a bug with how I implemented the jsonpath support for explain.

That being said, it should be noted that jsonpath in Kubernetes is not full jsonpath. I've been meaning to extend the library to be more fully compliant with the jsonpath standard but have not been able to do so yet. But the `..` should be acceptable in the library we have.

Reply:
Thanks. I confirmed that it still doesn't work with v1.28.3. For example:

~~~
$ kubectl explain pods..securityContext
KIND:       Pod
VERSION:    v1

error: field "securityContext" does not exist
~~~

Interestingly, I get a different error from `deployments` than `pods`. I'm not sure how to address it, though. It could be helpful for the output to include a legal example.

~~~
$ kubectl explain deployments..securityContext
error: invalid jsonpath syntax, first node must be field node
~~~

As a quick fix, all help text and error messages could drop references to jsonpath and simply note that the field names are to be joined by dots.

Beyond that it could be helpful to allow JSONPath. Currently `explain` doesn't seem to take more than one resource type. So treating the input as JSONPath may raise the question of what happens when more than one field is matched.

Something like the following could be quite useful in directing users toward the right command:

~~~
Multiple fields matched. Try one of the following:

kubectl explain --api-version apps/v1 Deployment.spec.template.spec.containers.securityContext
kubectl explain --api-version apps/v1 Deployment.spec.template.spec.ephemeralContainers.securityContext
kubectl explain --api-version apps/v1 Deployment.spec.template.spec.initContainers.securityContext
kubectl explain --api-version apps/v1 Deployment.spec.template.spec.securityContext
~~~

Or something similar to the --recursive output, but limited to matching fields and their ancestors:

~~~
GROUP:      apps
KIND:       Deployment
VERSION:    v1

DESCRIPTION:
    Deployment enables declarative updates for Pods and ReplicaSets.

MATCHING FIELDS:
  spec  <DeploymentSpec>
    template    <PodTemplateSpec> -required-
      spec      <PodSpec>
        containers      <[]Container> -required-
          securityContext       <SecurityContext>
        ephemeralContainers     <[]EphemeralContainer>
          securityContext       <SecurityContext>
        initContainers  <[]Container>
          securityContext       <SecurityContext>
        securityContext <PodSecurityContext>
~~~

Reply:
We should not bias to quick fixes that simply update the help text. I think I know what the problem is, but I will have to investigate further. I'll try to get to this ASAP since I'm the one that implemented this in the first place.

Reply:
Thank you for pursuing this